src/irmin/node.ml

```diff
@@ -363,6 +377,58 @@ struct
       | _ -> None)
 end
 
+module Make_generic_key_v2
+    (Hash : Hash.S) (Path : sig
+      type step [@@deriving irmin]
+    end)
+    (Metadata : Metadata.S)
+    (Contents_key : Key.S with type hash = Hash.t)
+    (Node_key : Key.S with type hash = Hash.t) =
+struct
+  include Make_generic_key (Hash) (Path) (Metadata) (Contents_key) (Node_key)
+
+  module Hash_preimage = struct
+    type entry =
+      | Node_hash of Hash.t node_entry
+      | Contents_hash of Hash.t contents_entry
+      | Contents_m_hash of Hash.t contents_m_entry
+    [@@deriving irmin]
+
+    type t = entry list [@@deriving irmin ~pre_hash]
+  end
+
+  let t =
```

Comment:
Was all that duplication necessary (it wasn't in `commit.ml`)? Would it be possible to prefix `t` directly instead of `Hash_preimage.t`?

Reply:
you are right, it wasn't